src/modules/app/StateType.js

```diff
@@ -14,13 +14,15 @@ export type CategoryRouteStateType = {|
                                                    (e.g. CategoriesStateType) to save memory
                                                    in the state. This would be an optimization! */
   +children: { [path: PathType]: Array<PathType> },
-  +allAvailableLanguages: Map<string, string> // including the current content language
+  +allAvailableLanguages: Map<string, string>, // including the current content language
+  +previousLanguage: string
```

Comment:
I did not really get why `previousLanguage` is needed. Can you explain this?

Reply:
I guess it's used to change the language back to the previous one, when navigating back (see clearCategory)